pkg/chunked/storage_linux.go

```diff
@@ -1280,6 +1280,13 @@ func typeToOsMode(typ string) (os.FileMode, error) {
 	return 0, fmt.Errorf("unknown file type %q", typ)
 }
 
+// cleanAbsDirectory removes any ".." and "." from the path
+// and ensures it starts with a "/".  If the path refers to the root
+// directory, it returns "/".
+func cleanAbsDirectory(path string) string {
```

Comment:
Nice, this is elegant.

---

- `s/cleanAbsDirectory/cleanAbsPath`, it is used for non-directories as well
- This change, effectively, changes the semantics of `fsVerityDigestsKey`. That’s fine, and AFAICS it works correctly — but it seems that changing `dump.sanitizeName` to this implementation, for consistency, would be nice. Actually, I think moving this function to `pkg/chunked/internal/composefs` (or a new subpackage of `…/internal`), and calling the function in both places, would be best.